### PR TITLE
Issue #664 - pre 3.6 strict option parsing behavior option

### PR DESCRIPTION
**Pull request by BlueTufa**
Maintain backward compatibility for pre-36 strict option parsing behavior for large existing code bases.

#664

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -705,6 +705,40 @@ Both these settings (`strictOptionParsing` and `strictArrayExtraction`) can be e
 val formats: Formats = DefaultFormats.strict
 ```
 
+Json 3.6 expands the `strictOptionParsing` behavior to throw a MappingException even when `extractOpt` is called on a missing optional attribute.  This behavior might not be desirable if you have a large code base that was using `extractOpt` prior to Json 3.6.  If you prefer to maintain the pre-3.6 `extractOpt` behavior, you can specify the format option `strictOptionParsingPre36`
```
Comment (mccartney) [inline README.md:708]: (A by-stander here)
Shouldn't the flag be called with respect to what it does. In two years "no-one" will care for what was the behavior before `3.6`

Comment (mccartney) [reply README.md:708]: @BlueTufa?

Comment (BlueTufa) [reply README.md:708]: How about `strictOptionParsingIgnoreMissing`?

**Comment by mccartney** [comment]
Looks fine to me. I need to consider this versus #686 (which solves a different problem). It seems like there's a potential for having a more consistent configuration experience.

**Comment by BlueTufa** [comment]
> Looks fine to me. I need to consider this versus #686 (which solves a different problem). It seems like there's a potential for having a more consistent configuration experience.

I am reviewing the proposal to #686 to see if it will meet our requirements.

**Comment by BlueTufa** [comment]
> > Looks fine to me. I need to consider this versus #686 (which solves a different problem). It seems like there's a potential for having a more consistent configuration experience.
> 
> I am reviewing the proposal to #686 to see if it will meet our requirements.

@mccartney I reviewed the proposed change #686 and the new configuration options align well with our requirements.  If you choose to proceed with #686 then I would refactor this PR to use the `nullExtractionStrategy .TreatAsAbsent` instead of adding a new value.

**Comment by magnolia-k** [comment]
@BlueTufa

#686 has been merged.

Do you plan to rewrite this PR?

**Comment by BlueTufa** [comment]
> @BlueTufa
> 
> #686 has been merged.
> 
> Do you plan to rewrite this PR?

@magnolia-k I will review it.  thank you!

**Pull request closed without merging**